great_expectations_provider/operators/great_expectations.py

```diff
@@ -44,121 +78,404 @@ class GreatExpectationsOperator(BaseOperator):
+    :param checkpoint_kwargs: A dictionary whose keys match the parameters of CheckpointConfig which can be used to
+        update and populate the Operator's Checkpoint at runtime
     :type checkpoint_kwargs: Optional[Dict]
-    :param fail_task_on_validation_failure: Fail the Airflow task if the Great Expectation validation fails
-    :type fail_task_on_validation_failure: Optiopnal[bool]
     :param validation_failure_callback: Called when the Great Expectations validation fails
     :type validation_failure_callback: Callable[[CheckpointResult], None]
+    :param fail_task_on_validation_failure: Fail the Airflow task if the Great Expectation validation fails
+    :type fail_task_on_validation_failure: bool
     :param return_json_dict: If True, returns a json-serializable dictionary instead of a CheckpointResult object
     :type return_json_dict: bool
-    :param **kwargs: kwargs
-    :type **kwargs: Optional[dict]
+    :param use_open_lineage: If True (default), creates an OpenLineage action if an OpenLineage environment is found
+    :type use_open_lineage: bool
     """
 
     ui_color = "#AFEEEE"
     ui_fgcolor = "#000000"
-    template_fields = (
-        "data_context_root_dir",
-        "checkpoint_name",
-        "checkpoint_kwargs",
-    )
+
+    operator_extra_links = (GreatExpectationsDataDocsLink(),)
 
     def __init__(
         self,
-        *,
         run_name: Optional[str] = None,
+        conn_id: Optional[str] = None,
+        execution_engine: Optional[str] = None,
+        expectation_suite_name: Optional[str] = None,
+        data_asset_name: Optional[str] = None,
         data_context_root_dir: Optional[Union[str, bytes, os.PathLike]] = None,
         data_context_config: Optional[DataContextConfig] = None,
+        dataframe_to_validate: Optional[
+            DataFrame
+        ] = None,  # should we allow a Spark DataFrame as well?
+        query_to_validate: Optional[str] = None,
         checkpoint_name: Optional[str] = None,
         checkpoint_config: Optional[CheckpointConfig] = None,
-        checkpoint_kwargs: Optional[Dict] = None,
-        fail_task_on_validation_failure: Optional[bool] = True,
+        checkpoint_kwargs: Optional[Dict[str, Any]] = None,
         validation_failure_callback: Optional[
             Callable[[CheckpointResult], None]
         ] = None,
+        fail_task_on_validation_failure: bool = True,
         return_json_dict: bool = False,
-        **kwargs
-    ):
-        super().__init__(**kwargs)
+        use_open_lineage: bool = True,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
 
+        self.data_asset_name: Optional[str] = data_asset_name
         self.run_name: Optional[str] = run_name
+        self.conn_id: Optional[str] = conn_id
+        self.execution_engine: Optional[str] = (
```

Comment:
Might it be better to require a valid `ExecutionEngine` and not default to `Pandas` (but raise an error if `None` or illegal `ExecutionEngine` is specified)?  Thanks.